src/js/game/LevelMVC/LevelPlane.js

```diff
@@ -87,12 +87,51 @@ module.exports = class LevelPlane extends Array {
   * Changes the block at a desired position to the desired block.
   * Important note: This is the cornerstone of block placing/destroying.
   */
-  setBlockAt(position, block) {
+  setBlockAt(position, block, direction = null) {
```

Comment:
Can we match the signature of `getBlockAt(position, offsetX = 0, offsetY = 0)` here?

Reply:
This is... do-able, yeah. The problem is "direction" as it exists in the controller (which again, this is all to do with that -dumb- plate-next-to-door case) is literally just 0,1,2, or 3. It's simply an int, that goes clockwise (so like if you're going up, direction == 0. Down and direction == 2).

I could, in the controller, evaluate the direction, case out the offset, and pass that in, but it seemed a bit weird to do that in the controller? I guess it could make sense to do it there, since that's really the only time direction matters, which makes doing the evaluation in setBlockAt() sort of weird...er

But turning a direction of 3 into an offset of [-1,0] would probably -have- to exist in one of the 2 functions.

Reply:
Ah, I see — in that case I think that's reasonable for now.